pkg/source/mtadapter/adapter.go

```diff
@@ -81,15 +79,19 @@ func NewAdapter(ctx context.Context, processed adapter.EnvConfigAccessor, ceClie
 func newAdapter(ctx context.Context, processed adapter.EnvConfigAccessor, ceClient cloudevents.Client, adapterCtor adapter.MessageAdapterConstructor) adapter.Adapter {
 	logger := logging.FromContext(ctx)
 	config := processed.(*AdapterConfig)
+
 	ml := resource.MustParse(config.MemoryLimit)
+	if ml.Value() > 0 {
+		logger.Infof("Setting MaxResponseSize to %d bytes", int32(ml.Value()))
+		sarama.MaxResponseSize = int32(ml.Value())
```

Comment:
Reading the doc, I saw this: 
```
The implementation on the server side would grab data from each partition in the fetch request until it hit this limit, then send back just the data for the partitions that fit in the response. The implementation would need to start from a random position in the list of topics included in the fetch request to ensure that in a case of backlog we fairly balance between partitions (to avoid first giving just the first partition until that is exhausted, then the next partition, etc).
```
It seems like the this MaxResponseSize is for all the messages in different partitions the broker returns all at once. Am I understanding this correctly? Currently this value comes from `VREPLICA_LIMITS_MEMORY`, which is set at only 2Mi. Should this number be changed as well? 2Mi seems very small.

Reply:
Yes that's my understanding as well. MaxResponseSize should be VREPLICA_LIMITS_MEMORY * # of vreplicas. Fixing.. Thanks for catching this!